### PR TITLE
Use _geoBoundingBox when creating the geo search filter

### PR DESCRIPTION
**Pull request by bidoubiwa**
As per the specification: https://github.com/meilisearch/specifications/pull/223

We used to transform the `insideBoundingBox` provided by `instantsearch` to a filter using `_geoRadius`. This required a lot of maths and was not the behavior expected by the user.

With the introduction of the `_geoBoundingBox` filter in Meilisearch, we can now use that instead of `_geoRadius`. 

⚠️ Search results may differ as the geo search algorithm changed

### CODE REVIEW

**Review by alallema — APPROVED** [review]
LGTM!
Just one question

File: packages/instant-meilisearch/src/adapter/search-request-adapter/geo-filter-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  aroundRadius,
+  minimumAroundRadius,
+}: InstantSearchGeoParams): string | undefined {
+  let middlePoint: string[] | undefined
+  let radius: number | undefined
+  let filter: string | undefined
+
+  if (aroundLatLng) {
+    const [lat3Str, lng3Str] = aroundLatLng.split(',')
+    const lat3 = Number.parseFloat(lat3Str).toFixed(5)
```
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply packages/instant-meilisearch/src/adapter/search-request-adapter/geo-filter-adapter.ts:15]: Yes, it's [in the standard library](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed)

**Review by alallema — APPROVED** [review]

File: packages/instant-meilisearch/src/adapter/search-request-adapter/__tests__/geo-rules.test.ts [hunk 2]
```diff
@@ -1,82 +1,82 @@
-import { adaptGeoPointsRules } from '../geo-rules-adapter'
+import { adaptGeoFilter } from '../geo-rules-adapter'
 
-test('Adapt geoPoints rules without a boundingBox', () => {
-  const rules = adaptGeoPointsRules()
-  expect(rules).toBeUndefined()
+test('Adapt geoPoints filter without a boundingBox', () => {
```
Comment (mdubus) [inline packages/instant-meilisearch/src/adapter/search-request-adapter/__tests__/geo-rules.test.ts:3]: `geoPoints filter` or `geo filter` (as in `adaptGeoFilter`)? I'm confused now :D

File: packages/instant-meilisearch/src/adapter/search-request-adapter/search-params-adapter.ts [hunk 3]
```diff
@@ -176,15 +173,26 @@ export function MeiliParamsCreator(searchContext: SearchContext) {
         meiliSearchParams.sort = [sort]
       }
     },
-    addGeoSearchRules() {
-      const geoSearchContext = createGeoSearchContext(searchContext)
-      const geoRules = adaptGeoPointsRules(geoSearchContext)
+    addGeoSearchFilter() {
```
Comment (mdubus) [inline packages/instant-meilisearch/src/adapter/search-request-adapter/search-params-adapter.ts:176]: Same here, I'm confused with the naming. 
Upper you do the following:
`import { adaptGeoFilter } from './geo-rules-adapter'`
And here you're writing `addGeoSearchFilter`. In the code you sometimes have `geoXXX`, `geoPointsXXX` and `geoSearchXXX`. I feel like the `geoXXX` is no precise enough and we don't really know what it is exactly. 
Is this intended or is something missing in the naming?

File: packages/instant-meilisearch/src/adapter/search-request-adapter/__tests__/geo-rules.test.ts (hunk 2 above)
Comment (bidoubiwa) [reply packages/instant-meilisearch/src/adapter/search-request-adapter/__tests__/geo-rules.test.ts:3]: I agree, let me rename all the tests

**Review by mdubus — APPROVED** [review]
LGTM ✨🦕

**Pull request merged**